### PR TITLE
URL input polish

### PR DESCRIPTION
**Pull request by jasmussen**
This PR fixes two issues with the URL input dialog.

The first issue is that the URL bar, in very tight circumstances, would create a horizontal scrollbar — because it was abs positioned at the center of where it needed to be and then pulled leftwards 50% — but the invisible div was still poking out at the right. First commit fixes this.

The second was a scoping issue with the URL input when used for the Image block, when toolbar is fixed to top. This fixes that:

screen shot 2018-04-20 at 10 17 36

### CODE REVIEW

File: blocks/rich-text/format-toolbar/index.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ class FormatToolbar extends Component {
 
 				{ ( isAddingLink || isEditingLink || formats.link ) && (
 					<Fill name="RichText.Siblings">
-						<div style={ { position: 'absolute', ...focusPosition } }>
+						<div style={ { position: 'absolute', transform: 'translateX( -50% )', ...focusPosition } }>
```
Comment (noisysocks) [inline blocks/rich-text/format-toolbar/index.js]: One sneaky `position` was OK, but now that there are a few properties I think we ought to give this div a `className` and add a new rule to `format-toolbar/style.scss`:

```scss
.blocks-format-toolbar__link-container {
    position: absolute;
    transform: translateX( -50% );
}
```

File: blocks/rich-text/format-toolbar/style.scss [hunk 2]
```diff
@@ -14,6 +14,12 @@
 	font-size: $default-font-size;
 	line-height: $default-line-height;
 	z-index: z-index( '.blocks-format-toolbar__link-modal' );
+
+	.edit-post-header-toolbar__block-toolbar & {
+		position: absolute;
+    	left: 0;
```
Comment (noisysocks) [inline blocks/rich-text/format-toolbar/style.scss]: There's some weird indentation going on here.

**Review by noisysocks — CHANGES_REQUESTED** [review]
Thanks for fixing this! ❤️ 

The UI shown when adding a link to an image looks and behaves 💯.

I do see one regression, though. When inserting a link into a paragraph block, the UI moves around from side to side a lot while you are typing:

links

**Comment by jasmussen** [comment]
Thank you for the feedback! I believe I addressed it:

screen shot 2018-04-23 at 11 59 58

File: blocks/rich-text/format-toolbar/style.scss [hunk 3]
```diff
@@ -17,11 +16,17 @@
...
 	.edit-post-header-toolbar__block-toolbar & {
 		position: absolute;
-    	left: 0;
+		left: 0;
 		right: 0;
 	}
 }
 
+.blocks-format-toolbar__link-container {
+    position: absolute;
```
[the remark this answers is not in the document]
Comment (jasmussen) [reply blocks/rich-text/format-toolbar/style.scss]: Hmm interesting — I was sure I installed eslint in VS Code — wondering why it didn't catch this...

**Review by noisysocks — APPROVED** [review]

**Comment by jasmussen** [comment]
Hooray! Thanks!

**Pull request merged**